### ISSUE TITLE
Outdated provider signature w/ Buzzard

### ISSUE THREAD

**Issue by alexisabril** [feathersjs-ecosystem/feathers-swagger#83]
`feathers-swagger` uses the Auk & prior signature to create a new provider:

```
providers.push((path, service) => {});
```

Usage:
https://github.com/feathersjs-ecosystem/feathers-swagger/blob/master/src/index.js#L76

Reference in Auk:
https://github.com/feathersjs/feathers/blob/3756f99af7680b6e9e77b70d6e2f46efba1a6839/src/application.js#L49

This breaks in Buzzard, due to the swapping of `path` and `service`:

```
providers.push((service, path) => {});
```

Reference in Buzzard:
https://github.com/feathersjs/feathers/blob/d4688ca9874b242a9b453caddf117af27f8a0b01/lib/application.js#L115

A fix is to just swap the two parameters, however, I'm unsure how the Feathers team is handling 2.0 -> 3.0 breaking changes.

**Comment by alexisabril** [comment]
From a discussion with @daffl: https://feathersjs.slack.com/archives/G62JKS5GF/p1510012938000055

```
const configurePlugin = (service, path) => {
  service.docs = service.docs || {};

  // Load documentation from service, if available.
  const doc = service.docs;

  // ...
}

if(app.version && app.version >= 3.0.0) {
  app.mixins.push(configurePlugin);
} else {
  app.providers.push((path, service) => configurePlugin(service, path));
}
```

**Comment by daffl** [comment]
Closed via #84
